lib/matplotlib/widgets.py

```diff
@@ -255,7 +255,7 @@ class Slider(AxesWidget):
     def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
                  closedmin=True, closedmax=True, slidermin=None,
                  slidermax=None, dragging=True, valstep=None,
-                 orientation='horizontal', **kwargs):
+                 valsnap=None, orientation='horizontal', **kwargs):
```

Comment:
If you are adding a new argument it has to go at the end (or better yet be keyword only!).  This in an API break as if there is someone (admittedly making a poor coding choice) of passing in 13 positional arguments, `valsnap` would get the string intended for `orientation` and likely blow up.

Reply:
Yikes. But yes ok, I suppose having millions of users nearly guarantees that that is happening somewhere.